### PR TITLE
chaincfg: Rework init time deployment val logic.

### PR DESCRIPTION
**Pull request by davecgh**
This reworks the deployment validation logic that runs at initialization time to make it easier to modify and bring it more in line with the more modern coding practices used in the codebase.  Similarly, it also reworks the associated deployment validation tests to make them more consistent with modern practices in the code as well as to simplify them and make them more flexible by moving the primary logic for all of the expected results into the test definitions themselves.

The following is a high-level overview of the changes:

- Use `math/bits` from the `stdlib` for population and trailing zero counts
- Rework error handling to use wrapped errors for better error messages while still allowing testability
- Rename several internal errors to better describe what they represent
- Add additional checks for missing abstain and no choices
- Modify `validateDeployments` to accept all deployments for all versions
- Return errors from `validateDeployments`
- Only `panic` directly in the `init` func
- Add missing comments and make the existing ones match reality
- Rework associated deployment validation tests
- Adds test for the missing abstain and missing no choice cases

### CODE REVIEW

File: chaincfg/init_test.go [hunk 1]
```diff
@@ -11,373 +11,175 @@ import (
...
+		deployments: map[uint32][]ConsensusDeployment{
+			1: {
+				{Vote: mockVote()},
+				{Vote: mockVote()},
+			},
 		},
-	}
+		expected: errDuplicateVoteId,
+	}, {
+		name: "allow duplicate vote id in the different versions",
```
Comment (davecgh) [inline chaincfg/init_test.go:171]:
```suggestion
		name: "allow duplicate vote id in different versions",
```

File: chaincfg/init.go [hunk 2]
```diff
@@ -105,73 +82,71 @@ func validateChoices(mask uint16, choices []Choice) error {
...
 		}
 		dups[id] = struct{}{}
 	}
 
-	// Check that there is only one IsNo and IsAbstain flag set to true.
-	if numAbstain != 1 {
-		return errInvalidIsAbstain
+	// Ensure there is one and only one of each choice type identification flag
+	// set.
+	if numAbstain == 0 {
```
Comment (davecgh) [inline chaincfg/init.go:95]: This would cleaner as a switch statement.

File: chaincfg/init_test.go (hunk 1 above)
Comment (davecgh) [reply chaincfg/init_test.go]: Updated.

File: chaincfg/init.go (hunk 2 above)
Comment (davecgh) [reply chaincfg/init.go]: Changed.

**Review by JoeGruffins — APPROVED** [review]

**Pull request merged**